### PR TITLE
Check for CTFE imports into generic repo.

### PR DESCRIPTION
**Pull request by Martin2112**
As discussed earlier. A proposal to ensure we don't accidentally bring CT related code into the generic repo.

### CODE REVIEW

File: scripts/check_unforked.sh [hunk 1]
```diff
@@ -14,6 +14,16 @@ check_import() {
   fi
 }
 
+check_blacklist() {
+  local path="$1"
+
+  local result=$(grep -Hne '\("github.com/google/certificate-transparency-go/trillian/ctfe.*"\)' "$path")
+  if [[ ! -z "${result}" ]]; then
+    echo "$result - importing CTFE into generic code is not allowed"
```
Comment (daviddrysdale) [inline scripts/check_unforked.sh]: maybe a smidgeon more specific: "importing CT-specific code into generic Trillian code is not allowed"

File: scripts/check_unforked.sh [hunk 2]
```diff
@@ -14,6 +14,16 @@ check_import() {
   fi
 }
 
+check_blacklist() {
+  local path="$1"
+
+  local result=$(grep -Hne '\("github.com/google/certificate-transparency-go/trillian/ctfe.*"\)' "$path")
```
Comment (daviddrysdale) [inline scripts/check_unforked.sh]: maybe disallow anything under `c-t-go/trillian/` ?

File: scripts/check_unforked.sh (hunk 1 above)
Comment (Martin2112) [reply scripts/check_unforked.sh]: Done.

File: scripts/check_unforked.sh (hunk 2 above)
Comment (Martin2112) [reply scripts/check_unforked.sh]: Done.

**Comment by Martin2112** [comment]
Sorry, I saw this about 30 seconds after merging it. I think a whitelist is
harder as the blacklist just does an egrep for lines that match.

**Comment by Martin2112** [comment]
Ah, no I merged the other one. Too many PRs. I'll hold off for a bit.

**Comment by daviddrysdale** [comment]
If we shift to a whitelist with `tls` it would also need `asn1` at some point.  And until the `examples/ct/` directory migrates to trillian-examples there would be other things needed too.  So I think a blacklist is OK for now.

**Comment by Martin2112** [comment]
OK I think I'll merge this then and we can revisit it if necessary.

**Pull request merged**